benchmarks/encoder/scenario.py

```diff
@@ -0,0 +1,97 @@
+import os
+import random
+import string
+
+import pyperf
+
+from ddtrace.internal.encoding import Encoder
+from ddtrace.span import Span
+
+
+try:
+    # the introduction of the buffered encoder changed the internal api
+    # see https://github.com/DataDog/dd-trace-py/pull/2422
+    from ddtrace.internal._encoding import BufferedEncoder
+
+    def _init_encoder(max_size=8 << 20, max_item_size=8 << 20):
+        return Encoder(max_size, max_item_size)
+
+
+except:
+
+    def _init_encoder():
+        return Encoder()
+
+
+VARIANTS = [
+    dict(ntraces=1, nspans=1000),
+    dict(ntraces=100, nspans=1000),
+    dict(ntraces=1, nspans=1000, ntags=1, ltags=16),
+    dict(ntraces=1, nspans=1000, ntags=48, ltags=64),
+    dict(ntraces=1, nspans=1000, nmetrics=1),
+    dict(ntraces=1, nspans=1000, nmetrics=48),
+]
+
+
+def _rands(size=6, chars=string.ascii_uppercase + string.digits):
```

Comment:
I don't think it should matter much here but was there a reason you chose to use string.ascii_uppercase instead of string.ascii_letters?

Reply:
No special reason. I copied this over from a test of the encoder.

https://github.com/majorgreys/dd-trace-py/blob/5a1c80f7ec927960e1c4732e4fb88316021f0ac7/tests/tracer/test_encoders.py#L31-L32